### PR TITLE
chore: add pr and issue templates

### PR DESCRIPTION
**Pull request by Nytelife26**
## This relates to...

Objectives 1 and 2 of #584 (I also checked Objective 6, so that may be crossed
off too).

## Rationale

N/A (justified by above).

## Changes

- Added bug report issue template
- Added feature request issue template
- Added general PR template (the one used to make this PR)

### Features

N/A.

### Bug Fixes

N/A.

### Breaking Changes and Deprecations

N/A.

## Status

- [S] Tested
- [S] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [x] In review
- [ ] Merge ready

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## This relates to...
```
Comment (mcollina) [inline .github/PULL_REQUEST_TEMPLATE/template.md:1]: You'd need to add:

```
<!--
Before submitting a pull request, please read
https://github.com/nodejs/undici/blob/master/CONTRIBUTING.md.

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `npm test` passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->
```

File: .github/PULL_REQUEST_TEMPLATE/template.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Status
+
+KEY: S = Skipped, x = complete
+
+- [ ] Tested
+- [ ] Benchmarked (**optional**)
+- [ ] Documented
+- [ ] Review ready
+- [ ] In review
+- [ ] Merge ready
```
Comment (mcollina) [inline .github/PULL_REQUEST_TEMPLATE/template.md:36]: I would also add a checkbox like

```
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md#developers-certificate-of-origin-11)
```

**Review by mcollina — COMMENTED** [review]
Thanks! for this, I've added a few notes

**Comment by Nytelife26** [comment]
> Thanks! for this, I've added a few notes

All requested changes implemented, ready for re-review :)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ronag** [comment]
Thanks!

**Comment by Ethan-Arrowood** [comment]
Opened and merged before I could even review and it looks great 😄  thanks for adding these!

**Comment by Nytelife26** [comment]
> Thanks!

> Opened and merged before I could even review and it looks great smile thanks for adding these!

Happy to help :)

**Pull request merged**